src/runtime.cpp

```diff
@@ -84,7 +84,11 @@ namespace hpx
 #if defined(HPX_HAVE_STACKTRACES)
             if (verbosity >= 1)
             {
-                std::cerr << "{stack-trace}: " << hpx::util::trace() << "\n";
+                std::size_t const trace_depth =
+                    util::from_string<int>(get_config_entry(
+                        "hpx.trace_depth", HPX_HAVE_THREAD_BACKTRACE_DEPTH));
+                std::cerr << "{stack-trace}: " << hpx::util::trace(trace_depth)
+                          << "\n";
```

Comment:
I think this change needs to be applied in `custom_exception_info.cpp` as well to have it used for exceptions.

Reply:
Yep, I just applied this change there as well. Things should be better now.